app/src/test/java/org/mozilla/fenix/library/historymetadata/controller/HistoryMetadataGroupControllerTest.kt

```diff
@@ -158,13 +167,13 @@ class HistoryMetadataGroupControllerTest {
 
     @Test
     fun handleDelete() = testDispatcher.runBlockingTest {
-        controller.handleDelete(setOf(mozillaHistoryMetadataItem, firefoxHistoryMetadataItem))
+        controller.handleDelete(getMetadataItemsList().toSet())
 
         coVerify {
-            store.dispatch(HistoryMetadataGroupFragmentAction.Delete(mozillaHistoryMetadataItem))
-            store.dispatch(HistoryMetadataGroupFragmentAction.Delete(firefoxHistoryMetadataItem))
-            historyStorage.deleteHistoryMetadata(mozillaHistoryMetadataItem.historyMetadataKey)
-            historyStorage.deleteHistoryMetadata(firefoxHistoryMetadataItem.historyMetadataKey)
+            getMetadataItemsList().forEach {
```

Comment:
You should also add a check that we're dispatching `DisbandSearchGroupAction` when appropriate.

e.g. for a deletion there are three conditions to check:
- have 2 items, deleting one;
- have 2 items, deleting all
- have 2 items, deleting 2

also, worth checking abnormal conditions to be a bit defensive, to make sure we don't blow up:
- have 2 items, deleting 5